src/Sulu/Bundle/MediaBundle/DependencyInjection/StorageCompilerPass.php

```diff
@@ -0,0 +1,217 @@
+<?php
+/*
+ * This file is part of the Sulu CMS.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\MediaBundle\DependencyInjection;
+
+use Sulu\Bundle\MediaBundle\Media\Exception\StorageAdapterNotFoundException;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\DefinitionDecorator;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Compiler pass for collecting services tagged with sulu_media.image.command.
+ */
+class StorageCompilerPass implements CompilerPassInterface
+{
+    /**
+     * @var array
+     */
+    protected $adapters = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition('sulu_media.storage_manager')) {
+            return;
+        }
+
```

Comment:
#L170 flysystem services are only created when flysystem classes exist. but still the storage manager need the rest of the code that it has the local storage.